Mathlib/Logic/Relation.lean

```diff
@@ -448,6 +448,18 @@ theorem _root_.WellFounded.transGen (h : WellFounded r) : WellFounded (TransGen
   ⟨fun a ↦ (h.apply a).TransGen⟩
 #align well_founded.trans_gen WellFounded.transGen
 
+section reflGen
+
+lemma reflGen_eq_self [IsRefl α r] : ReflGen r = r := by
```

Comment:
I'm not too happy with how this takes `IsRefl` but a later lemma takes `Reflexive r`. I think convention for this file is to ignore the typeclasses.

Reply:
```suggestion
lemma reflGen_eq_self (hr : IsTransitive r) : ReflGen r = r := by
```